DSharpPlus/Logging/DefaultLogger.cs

```diff
@@ -0,0 +1,88 @@
+﻿using System;
+using Microsoft.Extensions.Logging;
+
+namespace DSharpPlus
+{
+    public class DefaultLogger : ILogger<BaseDiscordClient>
+    {
+        private static readonly object _lock = new object();
+
+        private LogLevel MinimumLevel { get; }
+        private string TimestampFormat { get; }
+
+        internal DefaultLogger(BaseDiscordClient client)
+            : this(client.Configuration.MinimumLogLevel, client.Configuration.LogTimestampFormat)
+        { }
+
+        internal DefaultLogger(LogLevel minLevel = LogLevel.Information, string timestampFormat = "yyyy-MM-dd HH:mm:ss zzz")
+        {
+            this.MinimumLevel = minLevel;
+            this.TimestampFormat = timestampFormat;
+        }
+
+        public void Log<TState>(LogLevel logLevel, EventId eventId, TState state, Exception exception, Func<TState, Exception, string> formatter)
+        {
+            if (!this.IsEnabled(logLevel))
+                return;
+
+            lock (_lock)
+            {
+                Console.Write($"[{DateTimeOffset.Now.ToString(this.TimestampFormat)}] [{eventId.Id,-4}/{eventId.Name,-12}] ");
+
+                switch (logLevel)
+                {
+                    case LogLevel.Trace:
+                        Console.ForegroundColor = ConsoleColor.Gray;
+                        break;
+
+                    case LogLevel.Debug:
+                        Console.ForegroundColor = ConsoleColor.Magenta;
+                        break;
+
+                    case LogLevel.Information:
+                        Console.ForegroundColor = ConsoleColor.Cyan;
+                        break;
+
+                    case LogLevel.Warning:
+                        Console.ForegroundColor = ConsoleColor.Yellow;
+                        break;
+
+                    case LogLevel.Error:
+                        Console.ForegroundColor = ConsoleColor.Red;
+                        break;
+
+                    case LogLevel.Critical:
+                        Console.BackgroundColor = ConsoleColor.Red;
+                        Console.ForegroundColor = ConsoleColor.Black;
+                        break;
+                }
+                var level = logLevel switch 
+                {
+                    LogLevel.Trace =>       "trace",
```

Comment:
I think the levels should still start with uppercase letters. ("trace" -> "Trace", etc).